ECLIPs/ECIP-1054.md

```diff
@@ -0,0 +1,81 @@
+- [EIP 211](https://eips.ethereum.org/EIPS/eip-211) (New opcodes: RETURNDATASIZE and RETURNDATACOPY)
+- [EIP 214](https://eips.ethereum.org/EIPS/eip-214) (New opcode STATICCALL)
+- [EIP 658](https://eips.ethereum.org/EIPS/eip-658) (Embedding transaction status code in receipts)
+
+### Rationale
+
+This protocol specification notably omits the scheduled features of the anticipated _Constantinople_ protocol upgrade, which would be expected to include various further EVM upgrades. The reasoning for this omission hinges on a hedge toward battle-testing of those changes in light of multiple delays of that fork ([here](https://medium.com/ethereum-cat-herders/a-post-mortem-report-the-constantinople-ethereum-hard-fork-postponement-dd780d7ae63d), a postmortem of the latest delay) due to security and implementation discrepancies.
+
+__Interoperability__: establishing and maintaining interoperable behavior between Ethereum clients is important for developers and end-user adoption, yielding benefits for all participating chains (e.g., ETH and ETC, Ropsten and Morden, Görli and Kotti).
+
+__On Immutability__: Introducing new opcodes in the EVM has the potential to change behavior of existing contracts; in the case where previously an arbitrary invalid bytecode series (yielding _invalid opcode_) would now be assigned a meaning, and thus could generate or return a value other than _invalid_. In essence, this means _"possibly making music where there was only noise before."_ There is a concern that this behavior change contradicts an essential promise of Immutability, since an existing failing smart contract is liable to become a succeeding (not failing) contract, albeit in a hypothetical case of extreme coincidence and gross misuse of an opcode. In counterargument to this concern are two critical points:
+
+1. Account states remain unchanged.
+2. The _Homestead_ upgrade established a precedent for this type of change, having introduced the `DELEGATECALL` opcode at block `1_150_000`.
+
+With these arguments in place, along with precedence and expectation for other continuing and varied consensus-impacting protocol upgrades (e.g., soft- and hard-forks), it follows that the definition of immutability is not extended to guarantee perfect consistency for future _behavior_ of historical account states, but only to only to guarantee the immutability of the account states themselves.
+
+Adding opcodes and precompiled contracts to the EVM increases its functionality, and should be considered a feature upgrade rather than a modification.
+
+### Implementation
+
+Adoption of the content of this ECIP requires a hard fork as it introduces changes that are not backwards compatible.
+
+The following clients with Ethereum Classic support implement the Byzantium features:
+
+- Multi Geth: all Byzantium features
+- Parity Ethereum: all Byzantium features
+- Geth Classic: partial support for EIP-658 receipt status code change
```

Comment:
This client should need same features as others, or?

```suggestion
- Geth Classic: all Byzantium features
```

Reply:
This reflects the current state, not the desired state. In general, this section is used to point out a reference implementation which can be used by other client developer teams.

Reply:
IMO this should just be removed then; a spec should not be a roster of supposed current implementations unless the implementation(s) is part of the spec.

Reply:
For example, are you sure Parity-Ethereum supports an EIP100 implementation that hasn't also been tangled up with EIP659 or EIP1234?